### ISSUE TITLE
Make it an option to halt ViewManagerEventDispatchers on exceptions

### ISSUE THREAD

**Issue by asgerhallas** [d60/Cirqus#68]
It would be nice for testing with ViewManagerEventDispatcher and DependentViewManagerEventDispatcher, if they could be set up to rethrow exceptions from the view on the parent thread. 

Would you mind a HaltOnExceptions() option for these two? Or how do you use these in tests?

**Comment by asgerhallas** [comment]
Has anyone ever used InMemoryViewEventDispatcher? I could use something like that, but it does not seem to give access to the views, so I'm not really sure what the point is. Can I rewrite it without anyone gets hurt?

**Comment by mookid8000** [comment]
**Exceptions**: Could the worker thread raise an event on exceptions?

This way, the test context could subscribe to that and do whatever it feels like.

**In-mem view event dispatcher**: Never used it. Don't think anyone has. I promise I will not get hurt if you change it. (I think it's another story with `InMemoryViewManager` because it's pretty useful for doing in-mem testing of views)

**Comment by asgerhallas** [comment]
I ended up implementing a SynchronousViewManangerEventDispatcher for the testing utils, that works pretty well, and was easier than I anticipated. My fear was that supporting views with dependencies to other views would make require me to have an almost parallel implementation of both `ViewManagerEventDispatcher` and `DependentViewManagerEventDispacher`, but as it is synchronous, I just run through the give ViewManagers in the right order. Silly me. It's live on 0.63.4 :)
